test/io/nonblock/test_enumerator.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+require 'test/unit'
+require 'io/nonblock'
+require 'fiber'
+
+require_relative 'selector'
+
+class TestIOEnumerator < Test::Unit::TestCase
+  MESSAGE = "Hello World"
+  
+  def test_read
+    message = String.new
+    
+    thread = Thread.new do
+      selector = Selector.new
+      Thread.current.selector = selector
+      
+      i, o = IO.pipe
+      i.nonblock = true
+      o.nonblock = true
+      e = i.to_enum(:each_char)
```

Comment:
@funny-falcon does this cover the use case that you were interested in?

Reply:
My interest is correctly working code :-) If Fiber.yield works without breaking scheduler, it is good for everyone.

Reply:
There are lots of definitions for "correctly working code" :-)

In this example, the scheduler uses `fiber.transfer` as you suggested, and it correctly works inside enumeration. I wanted to confirm that it show a solution to the problem you mentioned in async.